src/geometry.ts

```diff
@@ -0,0 +1,304 @@
+/**
+ * Given a CSS px value (e.g. '10px'), strips the px suffix and returns the value as a number.
+ */
+function pxToNumber(value: string): number {
+	return Number(value.slice(0, -2));
+}
+
+/**
+ * Given a number, returns a CSS px value (e.g. '10px')
+ */
+function numberToPx(value: number): string {
+	return value + 'px';
+}
+
+/**
+ * Returns an object satisfying the ClientRect interface that is the result of adjusting the given ClientRect
+ * by the given dimensions.  The resulting object will have larger dimensions if the adjustment parameters
+ * are positive, or smaller dimensions if they are negative.
+ */
+function adjustBox(box: ClientRect, top: number, right: number, bottom: number, left: number): ClientRect {
+	return {
```

Comment:
Should we be checking to make sure the width and height are non-negative, and that the right/bottom is >= left/top ?